### PR TITLE
Rework Dropdown Scroll

### PR DESCRIPTION
**Pull request by wacky6**
Relevant components: Select, Cascader, TimeSelect

Please make sure these boxes are checked before submitting your PR, thank you!

* [ ] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md)).
* [ ] Make sure you are merging your commits to `dev` branch.
* [ ] Add some descriptions and refer relative issues for you PR.

### 做了什么
重新设计 Select 弹出菜单：弹出前将列表滚动到位。
将以上设计添加到 Cascader 和 TimeSelect

参见：https://github.com/ElemeFE/element/issues/5430

### Before (Flicker)
before

### After (No Flicker)
after

### Demo
https://jsfiddle.net/whm0ys1y/

### CODE REVIEW

File: packages/cascader/src/menu.vue [hunk 1]
```diff
@@ -94,6 +95,19 @@
...
+      },
+      scrollMenu(menu) {
+        scrollIntoView(menu, menu.getElementsByClassName('is-active')[0]);
+      },
+      handleMenuEnter() {
+        this.$nextTick(() => this.getMenus().forEach(menu => this.scrollMenu(menu)));
+      },
+      getMenus() {
+        if (this.$refs.wrapper) {
+          return Array.prototype.slice.call(this.$refs.wrapper.getElementsByClassName('el-cascader-menu'));
```
Comment (baiyaaaaa) [inline packages/cascader/src/menu.vue]: why not use ref="menus" & $refs.menus to resolve?

**Comment by baiyaaaaa** [comment]
Hi~Please split your pr to 2 commit with for each component.

File: packages/cascader/src/menu.vue (hunk 1 above)
Comment (wacky6) [inline packages/cascader/src/menu.vue]: 我不太确定怎么在 jsx 下用数组形式拿到 menu。

Vue 目前好像只支持在 `v-for` 和 `ref` 同时使用的时候返回数组形式的 $ref

**Comment by wacky6** [comment]
PR已拆分，Rebase了

File: packages/cascader/src/menu.vue (hunk 1 above)
Comment (baiyaaaaa) [inline packages/cascader/src/menu.vue]: 加 `refInFor` 就可以了

Comment (wacky6) [inline packages/cascader/src/menu.vue]: 改了，多谢大佬。涨姿势了

File: packages/date-picker/src/panel/time-select.vue [hunk 2]
```diff
@@ -81,6 +83,7 @@
         } else if (this.maxTime && compareTime(val, this.maxTime) > 0) {
           this.$emit('pick');
         }
+        this.$nextTick(() => this.scrollToOption());
```
Comment (Leopoldthecoder) [inline packages/date-picker/src/panel/time-select.vue]: @wacky6  这行代码导致了这个问题：https://jsfiddle.net/yqfamyhw/1/

有什么原因需要在每次选择完后触发 `scrollToOption` 方法吗？

**Pull request merged**